### PR TITLE
Some modest clockcult balancing - vitality matrices, brass skewers, pressure sensors

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
balance: Vitality matrices now apply Ichorial Stain when reviving a dead servant, which prevents that servant from being revived again by vitality matrices for a full minute.
balance: Vitality matrices can't be placed adjacent to one another.
balance: Brass skewers now damage and stun cyborgs.
balance: Vitality matrices no longer drain people impaled on brass skewers.
balance: Pressure sensors have 5 health again (down from 25.)
/:cl:

Notes on each change, in order:
* Someone raised a good point that each crewman that dies fuels the servants more because they allow the servants to infinitely revive. This is still the case, but the servants don't have functionally infinite retries in a short window now; combined with the upcoming change to let dead servants be dragged across the barrier, this gives them much fewer second chances.
* Even without messages, matrix spam is kind of silly, so this makes it impossible without a checkerboard pattern.
* While hilarious, cyborgs being able to walk through a skewer maze to make it completely destroyed breaks the balance. To compensate, they take heavy damage (50) when crossing one, and are slightly stunned, but the skewer breaks immediately.
* Putting a vitality matrix under a brass skewer pretty much guaranteed instant death to anyone who walked on it, followed by them being dusted without a chance to even more. Now they won't function on people buckled to things, including brass skewers. This also includes roller beds and the like.
* Less pressure sensor health compensates for the cyborg change and prevents being able to make an immense hazard and time-waster that serves no danger to your team while still being frustratingly resilient.

Also adds a small framework that lets status effects have special examine texts that show up on the owners, currently for humans and cyborgs. This is a new variable called `examine_text` on the status effect, with instances of `SUBJECTPRONOUN` being replaced appropriately by he, she, it, etc. To put these changes in an examine, just add `status_effect_examines()` to to the overall message shown.

### CODE REVIEW

File: code/game/gamemodes/clock_cult/clock_effects/clock_sigils.dm [hunk 1]
```diff
@@ -314,21 +314,28 @@
...
+							to_chat(L.pulledby, "<span class='userdanger'>[L] was already revived recently by a vitality matrix! Wait a bit longer!</span>")
+						break
+					else
+						if(ghost)
+							ghost.reenter_corpse()
+						L.revive(1, 1)
+						var/obj/effect/temp_visual/ratvar/sigil/vitality/V = new /obj/effect/temp_visual/ratvar/sigil/vitality(get_turf(src))
+						animate(V, alpha = 0, transform = matrix()*2, time = 8)
+						playsound(L, 'sound/magic/staff_healing.ogg', 50, 1)
+						L.visible_message("<span class='warning'>[L] suddenly gets back up, [L.p_their()] body dripping blue ichor!</span>", "<span class='inathneq'>\"[text2ratvar("You will be okay, child.")]\"</span>")
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_effects/clock_sigils.dm]: You can move the `<span class='warning'>[L] suddenly gets back up, [L.p_their()] body dripping blue ichor!</span>` to the status effect and just make the `<span class='inathneq'>\"[text2ratvar("You will be okay, child.")]\"</span>` a to_chat()

File: code/game/gamemodes/clock_cult/clock_effects/clock_sigils.dm [hunk 2]
```diff
@@ -279,7 +279,7 @@
 	animate(src, alpha = 255, time = 10, flags = ANIMATION_END_NOW) //we may have a previous animation going. finish it first, then do this one without delay.
 	sleep(10)
 //as long as they're still on the sigil and are either not a servant or they're a servant AND it has remaining vitality
-	while(L && (!is_servant_of_ratvar(L) || (is_servant_of_ratvar(L) && (GLOB.ratvar_awakens || GLOB.clockwork_vitality))) && get_turf(L) == get_turf(src))
+	while(L && (!is_servant_of_ratvar(L) || (is_servant_of_ratvar(L) && (GLOB.ratvar_awakens || GLOB.clockwork_vitality))) && get_turf(L) == get_turf(src) || !L.buckled)
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_effects/clock_sigils.dm]: uh, what are you doing? this will short-circuit to activating if they aren't buckled at all, regardless of actual status.

File: code/game/gamemodes/clock_cult/clock_structures/traps/brass_skewer.dm [hunk 3]
```diff
@@ -48,16 +48,24 @@
...
-		squirrel.apply_damage(20, BRUTE, "chest")
-		mouse_opacity = MOUSE_OPACITY_OPAQUE //So players can interact with the tile it's on to pull them off
-		buckle_mob(squirrel, TRUE)
+		if(iscyborg(squirrel))
+			if(!squirrel.stat)
+				squirrel.visible_message("<span class='boldwarning'>A massive brass spike erupts from the ground, rending [squirrel]'s chassis but shattering into pieces!</span>", \
+				"<span class='userdanger'>A massive brass spike rips through your chassis and bursts into shrapnel in your casing!</span>")
+				squirrel.adjustBruteLoss(50)
+				squirrel.Stun(20)
+				addtimer(CALLBACK(src, .proc/take_damage, max_integrity), 1)
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_structures/traps/brass_skewer.dm]: the spike should actually go through the borg for a very brief moment before this timer happens.

File: code/game/gamemodes/clock_cult/clock_scriptures/scripture_scripts.dm [hunk 4]
```diff
@@ -70,6 +70,12 @@
 	quickbind = TRUE
 	quickbind_desc = "Creates a Vitality Matrix, which drains non-Servants on it to heal Servants that cross it."
 
+/datum/clockwork_scripture/create_object/vitality_matrix/check_special_requirements()
+	if(locate(object_path) in range(1, invoker))
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_scriptures/scripture_scripts.dm]: This isn't a checkerboard pattern, since you're using a square range instead of a diamond range. You should use circlerange or the range that's cardinal that I can never remember.

File: code/game/gamemodes/clock_cult/clock_effects/clock_sigils.dm (hunk 2 above)
Comment (Xhuis) [reply code/game/gamemodes/clock_cult/clock_effects/clock_sigils.dm]: that was meant to be an &&, dang it!

File: code/game/gamemodes/clock_cult/clock_structures/traps/brass_skewer.dm (hunk 3 above)
Comment (Xhuis) [reply code/game/gamemodes/clock_cult/clock_structures/traps/brass_skewer.dm]: I wasn't sure fluffwise if the spear was powerful enough to pierce an entire metal machine, but I suppose it could go either way.

File: code/modules/mob/living/silicon/robot/examine.dm [hunk 5]
```diff
@@ -6,6 +6,10 @@
 	var/obj/act_module = get_active_held_item()
 	if(act_module)
 		msg += "It is holding [icon2html(act_module, user)] \a [act_module].\n"
+	for(var/V in status_effects)
+		var/datum/status_effect/E = V
+		if(E.examine_text)
+			msg += "[replacetext(E.examine_text, "SUBJECTPRONOUN", "It")]\n"
```
Comment (ChangelingRain) [inline code/modules/mob/living/silicon/robot/examine.dm]: Robots actually respect pronouns correctly. Also, this text should be in base examine *technically* but examine code sucks. Make it a proc.

File: code/datums/status_effects/debuffs.dm [hunk 6]
```diff
@@ -479,3 +479,27 @@
...
+//Ichorial Stain: Applied to servants revived by a vitality matrix. Prevents them from being revived by one again until the effect fades.
+/datum/status_effect/ichorial_stain
+	id = "ichorial_stain"
+	status_type = STATUS_EFFECT_UNIQUE
+	duration = 600
+	examine_text = "<span class='warning'>SUBJECTPRONOUN is drenched in thick, blue ichor!</span>"
+	alert_type = /obj/screen/alert/status_effect/ichorial_stain
+
+/datum/status_effect/ichorial_stain/on_apply()
+	owner.visible_message("<span class='danger'>Blue ichor covers [owner]'s body as they rise!</span>", \
```
Comment (ChangelingRain) [inline code/datums/status_effects/debuffs.dm]: p_they

File: code/game/gamemodes/clock_cult/clock_scriptures/scripture_scripts.dm (hunk 4 above)
Comment (Xhuis) [reply code/game/gamemodes/clock_cult/clock_scriptures/scripture_scripts.dm]: I might have misphrased; it *is* intentional to prevent an adjacency checkboard pattern, so that people can go between the matrices without having to cross a bunch of them.

File: code/modules/mob/living/silicon/robot/examine.dm (hunk 5 above)
Comment (Xhuis) [reply code/modules/mob/living/silicon/robot/examine.dm]: The replacement of SUBJECTPRONOUN or other text with the correct pronoun for the mob?

File: code/game/gamemodes/clock_cult/clock_effects/clock_sigils.dm (hunk 2 above)
Comment (ChangelingRain) [reply code/game/gamemodes/clock_cult/clock_effects/clock_sigils.dm]: Also, if it's meant to specifically prevent impaling, then make it check for specifically that so that idiots with rollerchair-cuffed crew don't faff around wondering why it doesn't work.
And **put it in the changelog**

File: code/modules/mob/living/silicon/robot/examine.dm (hunk 5 above)
Comment (ChangelingRain) [reply code/modules/mob/living/silicon/robot/examine.dm]: No, the status effects showing up in examine.

File: code/game/gamemodes/clock_cult/clock_effects/clock_sigils.dm (hunk 2 above)
Comment (Xhuis) [reply code/game/gamemodes/clock_cult/clock_effects/clock_sigils.dm]: This change actually slipped my mind completely when I was writing the PR body, so I'll put that in now.

File: code/modules/mob/living/silicon/robot/examine.dm (hunk 5 above)
Comment (Xhuis) [reply code/modules/mob/living/silicon/robot/examine.dm]: Ohh, good point.

**Comment by AdamElTablawy** [comment]
Ichoral stain is unnecessary. Servants are usually far less numerous than crew and need the revives considering they're often going up against guncargo and mechs with spears and good armor. If crew manages to bumble into the matrices with the new nerfs you have set up, they deserve to be feeding servants revives.

**Comment by Xhuis** [comment]
There have been a lot of rounds lately where servants just keep getting up and up every time they die, and it makes them functionally immortal. I'll give it some thought, though.

**Comment by ChangelingRain** [comment]
Here is a slightly improved alert icon http://puu.sh/yHGC1/aed477e19a.dmi that is consistent with all other clockcult alert icons.

**Comment by Xhuis** [comment]
Thanks, I'll put it in.

**Comment by Potato-Masher** [comment]
On the subject of traps and trap activators, what exactly is the intended function of the repeaters? Last time I tried to use them they just activate the skewer/vent once and then the traps never turn off, making them no different than a lever.

**Comment by Xhuis** [comment]
When they're active, they send an activation signal every second. Hooked to a vent, it'll constantly enable and disable the vent. Currently it doesn't have any real use outside of that, but I added the trigger so that future traps (i.e. clockwork emitters which I'm still waiting on sprites for from @Armhulen REEE) can be toggled constantly.

**Comment by Potato-Masher** [comment]
Ah cool, that's what I was thinking they were for. Was a little disappointed when I found out that the skewers couldn't be hooked to a repeater to make a terrifying spike maze.

Now that I remember it, can... wait. This idea I have is easy enough that I could do it if I could figure out how to git. The only guides I have read have been hard to follow, so I might have to sit down for an hour or two to study this harder...

**Comment by Robustin** [comment]
One minute CD is almost pointless. 

Reebe fights drag out for several minutes and a minute is close to the minimum amount of time I would expect a cultist to get revived, re-oriented, head back to the front line, fight, get killed, recovered, then placed on top of a rune again. Only really niche scenario would have the cultist killed and revived multiple times in the same 60 second window.

**Comment by Xhuis** [comment]
The one minute cooldown is made to counter spamming them all over the floor so you can go to one and get back up as soon as you die. The checkboarding also counters that, but the cooldown gives fewer second chances in a short timespan. If it proves ineffective, I'll buff it.

File: code/game/gamemodes/clock_cult/clock_effects/clock_sigils.dm [hunk 7]
```diff
@@ -278,7 +278,7 @@
 	animate(src, alpha = 255, time = 10, flags = ANIMATION_END_NOW) //we may have a previous animation going. finish it first, then do this one without delay.
 	sleep(10)
 //as long as they're still on the sigil and are either not a servant or they're a servant AND it has remaining vitality
-	while(L && (!is_servant_of_ratvar(L) || (is_servant_of_ratvar(L) && (GLOB.ratvar_awakens || GLOB.clockwork_vitality))) && get_turf(L) == get_turf(src))
+	while(L && (!is_servant_of_ratvar(L) || (is_servant_of_ratvar(L) && (GLOB.ratvar_awakens || GLOB.clockwork_vitality))) && get_turf(L) == get_turf(src) && !L.buckled)
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_effects/clock_sigils.dm]: this should still check specifically for brass skewers, rather than any buckling. rollerchairs aren't an uncommon form of restraint.

Comment (Xhuis) [reply code/game/gamemodes/clock_cult/clock_effects/clock_sigils.dm]: I thought I stated, but I guess I messed up and forgot to. Rollerchairs etc. are intended to go off, as pressure sensors detect pressure from biological creatures directly. That's also why silicons etc. didn't trigger it (I think, I may have had a different reason at the time.)

**Comment by Robustin** [comment]
So exactly how are we supposed to deal with spear traps again?

Just smash up every tile in our path?

**Comment by Xhuis** [comment]
Pressure sensors are very weak, so yes, smashing them up is the intended way. Clearing a path is the best way to counter spear-hells.
